bindings/python/opendaq/include/py_opendaq/py_typed_reader.h

```diff
@@ -179,82 +181,129 @@ struct PyTypedReader
     template <typename ValueType, typename ReaderType>
     static inline py::array_t<ValueType> read(const ReaderType& reader, size_t count, [[maybe_unused]] size_t timeoutMs)
     {
-        size_t blockSize = 1;
+        size_t blockSize = 1, initialCount = count;
+        constexpr const bool isMultiReader = std::is_base_of_v<daq::MultiReaderPtr, ReaderType>;
+
         if constexpr (std::is_same_v<ReaderType, daq::BlockReaderPtr>)
         {
             reader->getBlockSize(&blockSize);
         }
+        if constexpr (isMultiReader)
+        {
+            daq::ReaderConfigPtr readerConfig = reader.template asPtr<daq::IReaderConfig>();
+            blockSize = readerConfig.getInputPorts().getCount();
+        }
```

Comment:
General question - can these not be cached on construction?

Reply:
What do you mean?